source/core/ExecutionContext.cpp

```diff
@@ -312,12 +312,11 @@ InstructionCore* ExecutionContext::SuspendRunningQueueElt(InstructionCore* nextI
 // ExecuteSlices - execute instructions in run queue repeatedly (numSlices at a time before breaking out and checking
 // timers), until all clumps are finished or tickCount time is reached.
 // See enum ExecSlicesResult for explanation of return values, or comments below where result is set.
-Int32 /*ExecSlicesResult*/ ExecutionContext::ExecuteSlices(Int32 numSlices, PlatformTickType tickCount)
+Int32 /*ExecSlicesResult*/ ExecutionContext::ExecuteSlices(Int32 numSlices, Int32 millisecondsToRun)
```

Comment:
I would rather keep "PlatformTickType tickCount" as is. For some of the embedded devices it is foreseeable to call ExecuteSlices with time resolution smaller than milliseconds. I would rather fix the C EntryPoint EggShell_ExecuteSlices() to call tm->TheExecutionContext()->ExecuteSlices() after converting from the input millisecondsToRun to tickCount.
Also, currently, there are callsites of ExecutionContext::ExecuteSlices() call that assume the second argument is nanoseconds and not milliseconds.

Reply:
I respectfully disagree; I am in favor of this change.   While I think it's fine to have a platform-specific tickcount internally, I don't think it's good to have it in a public API.  I think standardizing on milliseconds is cleaner, and I don't think there's a practical reason for a target to need to run for less than 1 ms on any real target Vireo is likely to support; Vireo is not a practical real-time target, and RT targets are the only ones where I would see such support as actually useful.

Reply:
hmm, I thought Paul was experimenting with microblaze and Vireo. Do you all know what the scope of that research was?

Reply:
@spathiwa  wrote ", I don't think it's good to have it in a public API. I think standardizing on milliseconds is cleaner," 
>> Yes, I am also ok with public API (in CEntryPoints.cpp) to have millisecond resolution as mentioned in my comment above. I am also saying let's keep internal as TickCount. I think we actually agree.

Reply:
He was. I do not know the scope; perhaps if he is listening he can chime in.
But if there is a need to have a higher-resolution time period for this argument, I would favor adding a separate entry point, which *explicitly* takes nanoseconds rather than PlatformTickCount type.  And that's easy enough to do whenever we need it in the future, it shouldn't block this change.

Reply:
One thing I want to keep consistent is that ExecuteSlices takes a time to run and returns a time to wait and those units and types are the same for both.

I also would not like to use PlatformTickType because both the unit and type vary across platforms which makes for a messy public api.

We should also keep one entry point that uses a type of Int32 because asm.js has very poor performance handling and manipulating Int64 types (won't be the case for wasm, but something we have to deal with now).

As long as we use a unit that can represent at least a couple of orders of magnitude above kMaxExecWakeUpTime and fits in an Int32, I'm good.

Reply:
Is my understanding right about "EggShell_ExecuteSlices defined in CEntryPoints.cpp is public API. ExecutionContext::ExecuteSlices defined in ExecutionContext.cpp is not a public API."? If so, I am not asking about changing EggShell_ExecuteSlices defined in CEntryPoints.cpp. If not, I would like to understand this better. With the current change, we are calling ExecutionContext::ExecuteSlices with 10 million milliseconds in one place -- which does not sound right.

Reply:
I think we need to have this discussion on person to better understand the consequences, meanwhile Milan is blocked, and I would like Milan to continue, so I asked him to submit his changes. We can revisit later and adjust as necessary.

Reply:
Talked to @sanmut and agreed:
* the time unit in and out should be consistent so we won't revert the change
* longer term the main.cpp should be consuming the CEntryPoints that other embedders would use as well
* We can add entry points to the public api in the future for higher resolution timing and change the internal representation as needed